google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/ITDatastoreTest.java

```diff
@@ -137,6 +139,46 @@
   private static final Entity ENTITY3 = Entity.newBuilder(ENTITY1).setKey(KEY3).remove("str")
       .set("null", NULL_VALUE).set("partial1", PARTIAL_ENTITY2).set("partial2", ENTITY2).build();
 
+  private <T> Iterator<T> getStronglyConsistentResults (Query scQuery, Query query) {
+    QueryResults<T> scResults = DATASTORE.run(scQuery);
+    List<T> scResultsCopy = makeResultsCopy(scResults);
+    Set<T> scResultsSet = new HashSet<>(scResultsCopy);
+
+    infiniteloop:
+    while(true) {
+      QueryResults<T> results = DATASTORE.run(query);
+      List<T> resultsCopy = makeResultsCopy(results);
+      if (!haveSameSize(scResultsCopy.iterator(), resultsCopy.iterator())) {
+        continue;
+      }
+      for (T res: resultsCopy)  {
+        if (! scResultsSet.contains(res)) {
+          continue infiniteloop;
+        }
+      }
+      return resultsCopy.iterator();
```

Comment:
(Complementary to the codacy-bot comment).
There is no need to use labels and the "continue label" statement here, as it is the last statement in the whole method.

In this particular case the while(true) loop is used solely as a "label: [...] goto label;" construct. This can be avoided, using containsAll method (its implementation does exactly same thing in exactly same way):
```
while(true) {
  QueryResults<T> results = DATASTORE.run(query);
  List<T> resultsCopy = makeResultsCopy(results);
  if (!haveSameSize(scResultsCopy.iterator(), resultsCopy.iterator())) {
    continue;
  }
  if(csResultsSet.containsAll(resultsCopy)) {
    return resultsCopy.iterator();
  }      
}
```